sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -276,6 +276,7 @@ class Analyzer(override val catalogManager: CatalogManager)
       ResolveAliases ::
       ResolveSubquery ::
       ResolveSubqueryColumnAliases ::
+      ResolveLateralJoin ::
```

Comment:
Since `ResolveLateralJoin` has a similar functionality with `ResolveReferences`, how about putting this rule just after `ResolveReferences`?